django/contrib/auth/views.py

```diff
@@ -53,6 +53,22 @@ def inner(*args, **kwargs):
     return inner
 
 
+def deprecate_extra_context(func):
+    """
+    Handle deprecation of the extra_context parameter of the views.
+    """
+    @functools.wraps(func)
+    def inner(*args, **kwargs):
+        if 'extra_context' in kwargs:
+            warnings.warn(
+                "Passing `extra_context` as a keyword argument is deprecated.",
```

Comment:
"The unused `extra_context` parameter for `logout_and_login` is deprecated."
Also, you can move this into the view rather than creating a decorator since it's only used once.